x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/MetadataAttribute.java

```diff
@@ -173,12 +173,15 @@ public static boolean isSupported(String name) {
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (false == super.equals(obj)) {
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass() || super.equals(o) == false) {
```

Comment:
Could we encapsulate this repeated part (it's the same for all inheritors of NamedExpression) in a (static?) method, or would that kill any gains that we got from inlining this?